File: JsRuBot/bot.js
const linq = require('linq'),
    readline = require('readline'),
    log = console.log,
    rl = readline.createInterface(process.stdin, process.stdout),
    botName = 'vblz/tanks:jsrubot';

const UserActionType = { Move: 0, Shoot: 1},
    Direction = {Left: 0, Up: 1, Right: 2, Down: 3},
    CellType = {Tank: 0, Barrier: 1, NotDestroyable: 2, Water: 3};

rl.on('line', (state) => {
    let userAction = [];
    let gameState = JSON.parse(state);
    let myTank = linq.from(gameState.ContentsInfo).firstOrDefault((c) => c.Type === CellType.Tank && c.UserId === botName).Coordinates;
    let enemyTank = linq.from(gameState.ContentsInfo).firstOrDefault((c) => c.Type === CellType.Tank && c.UserId !== botName).Coordinates;
    let dX = myTank.X - enemyTank.X;
    let dY = myTank.Y - enemyTank.Y;

    let direction = (Math.random() > 0.7) ? (Math.floor(Math.random() * 4)) : (Math.abs(dX) > Math.abs(dY)
        ? (dX < 0 ? Direction.Right : Direction.Left)
        : (dY < 0 ? Direction.Up : Direction.Down));

    let action = (Math.random() > 0.5) ? (UserActionType.Move) : (UserActionType.Shoot);

    userAction.push({Type: action, Direction: direction});
    log(userAction);
});

File: BotsExamples/PyRuBot/bot.py
#!/usr/bin/env python3
import random
import json

botName = 'vblz/tanks:pyrubot'

while True:
    state = input()
    gameState = json.loads(state)
    myTank = next(c for c in gameState["ContentsInfo"] if c["Type"] == 0 and c["UserId"] == botName)["Coordinates"]
    enemyTank = next(c for c in gameState["ContentsInfo"] if c["Type"] == 0 and c["UserId"] != botName)["Coordinates"]
    dX = myTank["X"] - enemyTank["X"]
    dY = myTank["Y"] - enemyTank["Y"]
    if random.randint(0, 100) < 70:
        direction = ((0 if dX < 0 else 2) if (abs(dX) > abs(dY)) else (1 if dY < 0 else 3))
    else:
        direction = (random.randint(0, 3))

    print([{'Type': 0 if random.randint(0, 100) < 70 else 1, 'Direction': direction}])

